### PR TITLE
cli: Comment and update helper function for retrieving kuberenets context to use

### PR DESCRIPTION
**Pull request by nfuden**
…
# Description

Update err handling in kubecontext on glooctl

## Code changes
- Fix error in gloo cli when dealing with kube context

## Notes for reviewers

Be sure to verify intended behavior by setting some options in your kubernetes context such as which cluster to use and validate with checkcrds or some other glooctl command of your choosing.

glooctl  checkcrds --kube-context my-context2

BOT NOTES: 
resolves https://github.com/solo-io/gloo/pull/7623

### CODE REVIEW

**Review by sam-heilbron — APPROVED** [review]

**Review by davidjumani — APPROVED** [review]
LGTM, just a minor nit

File: projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go [hunk 1]
```diff
@@ -35,10 +40,12 @@ func ContextAccessibleFrom(ctx context.Context) (ContextAccessible, error) {
 	return ContextAccessible{}, eris.New("No options set on current context")
 }
 
+// KubecontextFrom pulls the kube context if it was stuffed into the go context.
+// Swallows the no options error that is normally retrieved.
 func KubecontextFrom(ctx context.Context) (string, error) {
 	opts, err := ContextAccessibleFrom(ctx)
-	if err != nil {
+	if err == nil {
 		return opts.KubeContext, nil
```
Comment (davidjumani) [inline projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go:48]: nit: Would it be better to invert the return values to keep the `if err != nil` paradigm ?

**Review by davidjumani — DISMISSED** [review]
LGTM, just a minor nit

File: projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go [hunk 2]
```diff
@@ -35,10 +40,12 @@ func ContextAccessibleFrom(ctx context.Context) (ContextAccessible, error) {
 	return ContextAccessible{}, eris.New("No options set on current context")
 }
 
+// KubecontextFrom pulls the kube context if it was stuffed into the go context.
+// Swallows the no options error that is normally retrieved.
```
Comment (jenshu) [inline projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go:44]: if this error is always swallowed, should we just change `ContextAccessibleFrom` to not return an error? this seems to be the only place that func is called

Comment (nfuden) [reply projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go:44]: Yeah I like that idea

File: projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go [hunk 3]
```diff
@@ -26,19 +28,21 @@ type Consul struct {
...
-	opts, err := ContextAccessibleFrom(ctx)
-	if err != nil {
-		return opts.KubeContext, nil
-	}
-	return "", err
+// KubecontextFrom pulls the kube context if it was stuffed into the go context.
+// Returns an empty string if it was not set.
+func KubecontextFrom(ctx context.Context) string {
+	opts := ContextAccessibleFrom(ctx)
+	return opts.KubeContext // gaurenteed to be set
```
Comment (jenshu) [inline projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go:46]: correction: opts is guaranteed to be set, but KubeContext can be empty string?

File: projects/gloo/cli/pkg/cmd/check-crds/root.go [hunk 4]
```diff
@@ -135,10 +135,7 @@ func preprocessCRD(crd *apiextv1.CustomResourceDefinition) {
 // getCRDsInCluster gets all custom resources currently in the local cluster
 func getCRDsInCluster(ctx context.Context) ([]apiextv1.CustomResourceDefinition, error) {
 	crds := []apiextv1.CustomResourceDefinition{}
-	kubecontext, err := contextoptions.KubecontextFrom(ctx)
-	if err != nil {
-		return nil, err
-	}
+	kubecontext := contextoptions.KubecontextFrom(ctx)
 	out, err := cliutil.KubectlOut(nil, "get", "crd", "--context", kubecontext)
```
Comment (jenshu) [inline projects/gloo/cli/pkg/cmd/check-crds/root.go:139]: should only pass in `--context` if kubecontext is not empty

File: projects/gloo/cli/pkg/cmd/install/knative.go [hunk 5]
```diff
@@ -214,10 +214,7 @@ func checkKnativeInstallation(ctx context.Context, kubeclient ...kubernetes.Inte
 	if len(kubeclient) > 0 {
 		kc = kubeclient[0]
 	} else {
-		kubecontext, err := contextoptions.KubecontextFrom(ctx)
-		if err != nil {
-			return false, nil, err
-		}
+		kubecontext := contextoptions.KubecontextFrom(ctx)
 		kc = helpers.MustKubeClientWithKubecontext(kubecontext)
```
Comment (jenshu) [inline projects/gloo/cli/pkg/cmd/install/knative.go:218]: everywhere we call KubecontextFrom, we should make sure that empty kubecontext is handled appropriately

File: projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go [hunk 6]
```diff
@@ -26,19 +28,21 @@ type Consul struct {
...
 
-func KubecontextFrom(ctx context.Context) (string, error) {
-	opts, err := ContextAccessibleFrom(ctx)
-	if err != nil {
-		return opts.KubeContext, nil
-	}
-	return "", err
+// KubecontextFrom pulls the kube context if it was stuffed into the go context.
+// Returns an empty string if it was not set.
+func KubecontextFrom(ctx context.Context) string {
```
Comment (jenshu) [inline projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go:44]:
```suggestion
func KubeContextFrom(ctx context.Context) string {
```
nit: casing

Comment (nfuden) [reply projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go:44]: Id rather not change an exported functions name even if its not technically correct

File: projects/gloo/cli/pkg/cmd/install/knative.go (hunk 5 above)
Comment (nfuden) [reply projects/gloo/cli/pkg/cmd/install/knative.go:218]: yep it looks like its safe in all cases but its definitely cleaner to not have the --context added

File: projects/gloo/cli/pkg/cmd/check-crds/root.go (hunk 4 above)
Comment (nfuden) [reply projects/gloo/cli/pkg/cmd/check-crds/root.go]: fair. Updated

File: projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go (hunk 3 above)
Comment (nfuden) [reply projects/gloo/cli/pkg/cmd/options/contextoptions/contextoptions.go]: yeah the struct is always concrete so this field access is always safe

**Review by jenshu — APPROVED** [review]

**Review by davidjumani — APPROVED** [review]

**Review by jenshu — APPROVED** [review]

**Pull request merged**